File: models/activity_log.py
# models/activity_log.py — Pydantic schemas for activity log responses
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class ActivityLogResponse(BaseModel):
    """Schema returned to the client for activity log entries."""
    id: int
    action: str
    status: str
    detail: str | None = None
    file_path: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: flutter_client/lib/models/activity_log.dart
// models/activity_log.dart
// Mirrors FastAPI's ActivityLog schema — tracks background job results.
// Statuses: "pending" → "completed" / "failed"

class ActivityLog {
  final int id;
  final String action;
  final String status;
  final String? detail;
  final String? filePath;
  final DateTime createdAt;
  final DateTime updatedAt;

  const ActivityLog({
    required this.id,
    required this.action,
    required this.status,
    this.detail,
    this.filePath,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ActivityLog.fromJson(Map<String, dynamic> json) {
    return ActivityLog(
      id: json['id'] as int,
      action: json['action'] as String,
      status: json['status'] as String,
      detail: json['detail'] as String?,
      filePath: json['file_path'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  bool get isPending => status == 'pending';
  bool get isCompleted => status == 'completed';
  bool get isFailed => status == 'failed';
}
